File: ChatTest2/app/src/main/java/com/jmnl2020/chattest/RecyclerviewAdapter.java
package com.jmnl2020.chattest;

import android.content.Context;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.storage.FirebaseStorage;
import com.google.firebase.storage.StorageReference;

import java.util.ArrayList;

public class RecyclerviewAdapter extends RecyclerView.Adapter {

    Context context;
    ArrayList<CardviewItem> items;

    //받아올 이미지뷰 참조변수
    ImageView iv;

    public RecyclerviewAdapter(Context context, ArrayList<CardviewItem> items) {
        this.context = context;
        this.items = items;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        //뷰를 만들어주는 메소드

        View itemView = LayoutInflater.from(context).inflate(R.layout.item_cardview, parent, false);
        VH holder = new VH(itemView);

        iv= itemView.findViewById(R.id.cv_img);

        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        VH vh= (VH) holder;

      vh.cvTitle.setText(  items.get(position).cvTitle );
      vh.cvTitle.setText(  items.get(position).cvTitle );
      vh.cvTitle.setText(  items.get(position).cvTitle );

      Glide.with(context).load(items.get(position).cvImgfile).into(vh.cvImgfile);

    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class VH extends RecyclerView.ViewHolder {

        ImageView cvImgfile;
        TextView cvTitle;
        TextView cvName;
        TextView cvMsg;

        public VH(@NonNull View itemView) {
            super(itemView);

            cvTitle = itemView.findViewById(R.id.cv_title);
            cvName = itemView.findViewById(R.id.cv_name);
            cvMsg = itemView.findViewById(R.id.cv_msg);
            cvImgfile = itemView.findViewById(R.id.cv_img);

        }
    }

}

File: calendarTest/app/src/main/java/com/jmnl2020/calendartest/RecyclerViewAdapter.kt
package com.jmnl2020.calendartest

import android.view.ViewGroup
import java.util.*

class RecyclerViewAdapter (val mainActivity: MainActivity) : RecyclerView.Adapter<ViewHolderHelper>(){


        val baseCalendar = BaseCalendar()

        init {
            baseCalendar.initBaseCalendar {
                refreshView(it)
            }
        }

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolderHelper {
            val view = LayoutInflater.from(parent.context).inflate(R.layout.item_schedule, parent, false)
            return ViewHolderHelper(view)
        }

        override fun getItemCount(): Int {
            return BaseCalendar.LOW_OF_CALENDAR * BaseCalendar.DAYS_OF_WEEK
        }

        override fun onBindViewHolder(holder: ViewHolderHelper, position: Int) {

            if (position % BaseCalendar.DAYS_OF_WEEK == 0) holder.tv_date.setTextColor(Color.parseColor("#ff1200"))
            else holder.tv_date.setTextColor(Color.parseColor("#676d6e"))

            if (position < baseCalendar.prevMonthTailOffset || position >= baseCalendar.prevMonthTailOffset + baseCalendar.currentMonthMaxDate) {
                holder.tv_date.alpha = 0.3f
            } else {
                holder.tv_date.alpha = 1f
            }
            holder.tv_date.text = baseCalendar.data[position].toString()
        }

        fun changeToPrevMonth() {
            baseCalendar.changeToPrevMonth {
                refreshView(it)
            }
        }

        fun changeToNextMonth() {
            baseCalendar.changeToNextMonth {
                refreshView(it)
            }
        }

        private fun refreshView(calendar: Calendar) {
            notifyDataSetChanged()
            mainActivity.refreshCurrentMonth(calendar)
        }
}